### PR TITLE
ROR Project

### PR DESCRIPTION
**Pull request by keroloussamy**
**Splash screen**
  A simple page with the name of your app (yes, you need to choose one), and links to the sign up and log in pages.

**Sign up and log in pages**
  The user should be able to register in the app with full name, email and password (all mandatory).
  The user can log into the app using email and password.
  If the user is not logged in, they can't access pages that require the user to be logged in (all the pages described below).

**Home page (catagories page)**
  When the user logs in, they are presented with the categories page.
  For each category, the user can see their name, icon and the total amount of all the transactions that belongs to that category.
  When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.
  There is a button "add a new category" at the bottom that brings the user to the page to create a new category.

**Transactions page**
  For a given category, the list of transactions is presented, ordered by the most recent.
  At the top of the page the user could see the total amount for the category (sum of all of the amounts of the transactions in that category).
  There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction.
  When the user clicks on the "Back" button (<), the user navigates to the home page.

**"Add a new category" page**
  The user fills out a form to create a new category, indicating their name and icon (both mandatory).
  The user click (or taps) the "Save" button to create the new category, and is taken to the home page on success.
  When the user clicks on the "Back" button (<), the user navigates to the home page.

  **"Add a new transaction" page**
  The user fills out a form to create a new transaction with:
  name (mandatory)
  amount (mandatory)
  categories (mandatory at least one)
  The user click (or taps) the "Save" button to create the new category, and is taken to the transactions page for that category.
  When the user clicks on the "Back" button (<), the user navigates to the transactions page for that category.

### CODE REVIEW

**Review by elisha2kyakpo1 — CHANGES_REQUESTED** [review]
Hi 👋 @keroloussamy,

You have done a great job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
### To highlight
- The PR description and title looks professional ✔️
- There are no linter errors ✔️
- Commit messages looks good 👍 


## Required Changes ♻️
_Check the comments under the review._

- [ ] Kindly record another video, and highlight a feature or piece of code you are proud of, the one available was cut short and didn't talk about a feature/code you are proud of. Remember that this is a project requirement.
- [ ] Let's add a log out button, I wanted to create at least more than one account for testing purposes, but because I failed to log out I couldn't do that.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: db/schema.rb [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  # These are extensions that must be enabled in order to support this database
+  enable_extension "plpgsql"
+
+  create_table "entities", force: :cascade do |t|
+    t.string "name"
+    t.integer "amount"
+    t.datetime "created_at", null: false
+    t.datetime "updated_at", null: false
+    t.bigint "user_id", null: false
+    t.index ["user_id"], name: "index_entities_on_user_id"
```
[the remark this answers is not in the document]
Comment (Whoistolu) [reply db/schema.rb]: Well resolved 👍🏽

File: db/schema.rb [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    t.index ["group_id"], name: "index_entities_groups_on_group_id"
+  end
+
+  create_table "groups", force: :cascade do |t|
+    t.string "name"
+    t.string "icon"
+    t.datetime "created_at", null: false
+    t.datetime "updated_at", null: false
+    t.bigint "user_id", null: false
+    t.index ["user_id"], name: "index_groups_on_user_id"
```
[the remark this answers is not in the document]
Comment (Whoistolu) [reply db/schema.rb]: Well resolved 👍🏽

File: app/models/group.rb [hunk 3]
```diff
@@ -0,0 +1,7 @@
+class Group < ApplicationRecord
+  belongs_to :user
```
[the remark this answers is not in the document]
Comment (Whoistolu) [reply app/models/group.rb]: Well implemented 👍🏽

**Review by Whoistolu — APPROVED** [review]
### Hi @keroloussamy 

Great job implementing the recommended changes 👍🏽 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

# STATUS: APPROVED 🟢
**Congratulations! 🎉** 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**